seahorn/lib/linked_list_helper.c

```diff
@@ -64,89 +27,259 @@ static inline void init_short_aws_linked_list(struct aws_linked_list *list,
   }
 }
 
-void sea_nd_init_aws_linked_list_from_front(struct aws_linked_list *list,
-                                            size_t *length) {
+// if nd_voidp() is assigned to next, prev without assigning them sane
+// values, then compiler treats them as undefined.
+void initNode(struct aws_linked_list_node *node) {
+  assume(node);
+  struct aws_linked_list_node *dummy =
+      malloc(sizeof(struct aws_linked_list_node));
+  node->next = dummy;
```

Comment:
this seems very suspicious. Why should next and prev be assigned to some malloc?
The comment mentions something called `sane`, but I don't know what that concept means.
The comment also suggests that compiler treats `insane` values as `undefined`. What exactly does
this mean? 

Overall, this seems like a hack to patch some case that was not fully understood. Things like this are show stoppers. I can't accept this PR until this is either resolved or documented better.

Reply:
Fixed.